swift/Lottie.swift

```diff
@@ -0,0 +1,152 @@
+import CoreGraphics
+import CoreMedia
+
+import thorvg
+
+/// Errors that can occur while working with Lottie animations.
+public enum LottieError: Error {
+    case failedToLoadLottieFromString
+    case failedToLoadLottieFromPath
+    case failedToDrawLottieFrameOntoCanvas
+    case failedToInitialiseTvgEngine
+    case frameIndexOutOfBounds
+}
+
+/// Shorthand for the buffer type used to represent image pixel data.
+public typealias Buffer = UnsafeMutablePointer<UInt32>
+
+/// Shorthand for a pointer to an internal Lottie animation object
+private typealias Animation = OpaquePointer
+
+/// Shorthand for a pointer to an internal rendering canvas
+private typealias Canvas = OpaquePointer
+
+/// Object used to load and render Lottie frames.
+public class Lottie {
```

Comment:
maybe we need to add the width and height properties that we get from lottie file?

Reply:
Great find! As we discussed, there is currently only one type of size we care about: _render size_.

This is passed into the `render` function, and we then set the output of the Lottie frame to be that size. 

However, in the context of Canva, we need to have the original size of the Lottie to help calculate any potentially cropping rectangle we have. 

For now, I'll add these parameters, but I'm keen to try and figure out if we can get away without them.